docs/spec_draft.md

```diff
@@ -704,6 +705,133 @@ implementation-defined.
+same floating-point semantics and works as follows:
+  * `rfft(real_operand) = truncated_result` where
+  * `complex_operand[i] = (real_operand, 0)` for all `i`.
+  * `complex_result = fft(complex_operand)`.
+  * `truncated_result = complex_result[:(rank(complex_result) / 2 + 1)]`.
+
+(When the discrete Fourier transform is computed for real operands, the first
+`N/2 + 1` elements of the result unambiguously define the rest of the result,
+so the result of `rfft` is truncated to avoid computing redundant elements).
+
+For `fft_type = RFFT`, `result` is defined as the final result of a series of L
+computations where `L = size(fft_length)`. For example, for `L = 3`:
+  * `result1[i0, ..., :]` = `rfft(operand[i0, ..., :])` for all `i`.
+  * `result2[i0, ..., :, iR-1]` = `fft(result1[i0, ..., :, iR-1])` for all `i`.
+  * `result[i0, ..., :, iR-2, iR-1]` = `fft(result2[i0, ..., :, iR-2, iR-1])`
+    for all `i`.
+
+Finally, given the function `irfft` which has the same type signature and
+computes the inverse of `rfft`:
+
+For `fft_type = IRFFT`, `result` is defined as the inverse of the computations
+for `fft_type = RFFT`. For example, for `L = 3`:
+  * `result1[i0, ..., :, iR-2, iR-1]` = `ifft(operand[i0, ..., :, iR-2, iR-1])`
+    for all `i`.
+  * `result2[i0, ..., :, iR-1]` = `ifft(result1[i0, ..., :, iR-1])` for all `i`.
+  * `result[i0, ..., :]` = `irfft(result2[i0, ..., :])` for all `i`.
+
+### Inputs
+
+| Name         | Type                                         |
+|--------------|----------------------------------------------|
+| `operand`    | tensor of floating-point or complex type     |
+| `fft_type`   | constant of type `si64`                      |
+| `fft_length` | 1-dimensional tensor constant of type `si64` |
+
+### Outputs
+
+| Name     | Type                                     |
+|----------|------------------------------------------|
+| `result` | tensor of floating-point or complex type |
+
+### Constraints
+
+  * (C1) rank(`operand`) is in range [1, 3].
+  * (C2) The relationship between `operand` and `result` types varies:
+    * If `fft_type = FFT`, `operand` and `result` have the same type.
+    * If `fft_type = IFFT`, `operand` and `result` have the same type.
+    * If `fft_type = RFFT`, `operand` has float and `result` has complex type of
+    the same floating-point semantics. Shape of the innermost axis is reduced to
+    `fft_length[-1] // 2 + 1` if `fft_length[-1]` is a non-zero value, omitting
+    the reversed conjugate part of the transformed signal beyond the Nyquist
+    frequency.
+    * If `fft_type = IRFFT`, `operand` has complex and `result` has float type
+    of the same floating-point semantics. Shape of the innermost axis is
+    expanded to `fft_length[-1]` if `fft_length[-1]` is a non-zero value,
+    inferring the part of the transformed signal beyond the Nyquist frequency
+    from the reverse conjugate of the 1 to `fft_length[-1] // 2 + 1` entries.
```

Comment:
In C2, let's focus solely on element types (this is why one of the comments in the previous review proposed "the relationship between `operand` and `result` element types varies"), because shapes are taken care of below. More concretely:

```
  * (C2) The relationship between `operand` and `result` element types varies:
    * If `fft_type = FFT`, `element_type(operand)` and `element_type(result)` 
      have the same complex type.
    * If `fft_type = IFFT`, `element_type(operand)` and `element_type(result)` 
      have the same complex type.
    * If `fft_type = RFFT`, `element_type(operand)` is a floating-point type and 
      `element_type(result)` is a complex type of the same floating-point 
      semantics.
    * If `fft_type = RFFT`, `element_type(operand)` is a complex type and
      `element_type(result)` is a floating-point type of the same floating-point 
      semantics.
```